<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>591</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QStackedWidget" name="mainStackedWidget">
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="newPasswordPage">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="4" column="2">
         <widget class="QPushButton" name="enterNewPassButton">
          <property name="text">
           <string>Enter</string>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Password</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Confirm Password</string>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <spacer name="horizontalSpacer_7">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="2" colspan="2">
         <widget class="QLineEdit" name="confirmPasswordField">
          <property name="inputMethodHints">
           <set>Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::PasswordEchoOnEdit</enum>
          </property>
          <property name="placeholderText">
           <string>Re-type Password</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2" colspan="2">
         <widget class="QLineEdit" name="newPasswordField">
          <property name="inputMethodHints">
           <set>Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::PasswordEchoOnEdit</enum>
          </property>
          <property name="placeholderText">
           <string>Password</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="3" alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Please enter a new password:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <spacer name="horizontalSpacer_8">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="4">
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="loginPage">
       <layout class="QGridLayout" name="gridLayout">
        <item row="3" column="2">
         <widget class="QPushButton" name="enterPassButton">
          <property name="text">
           <string>Enter</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="4" column="2">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="1">
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="4" rowspan="3">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="1" colspan="3">
         <widget class="QLineEdit" name="enterPassField">
          <property name="echoMode">
           <enum>QLineEdit::PasswordEchoOnEdit</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="3" alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="enterPasslabel">
          <property name="text">
           <string>Enter the password to decrypt datafile</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="3">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
       <zorder>enterPassButton</zorder>
       <zorder>enterPassField</zorder>
       <zorder>horizontalSpacer</zorder>
       <zorder>verticalSpacer</zorder>
       <zorder>horizontalSpacer_2</zorder>
       <zorder>verticalSpacer_2</zorder>
       <zorder>enterPasslabel</zorder>
       <zorder>horizontalSpacer_3</zorder>
       <zorder>horizontalSpacer_4</zorder>
      </widget>
      <widget class="QWidget" name="tokenPage">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="3">
         <widget class="QPushButton" name="exportButton">
          <property name="text">
           <string>Export</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <spacer name="horizontalSpacer_9">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="removeEntryButton">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="addEntryButton">
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="4">
         <widget class="QListWidget" name="listWidget"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="addEntryPage">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="9" column="2">
         <widget class="QRadioButton" name="sha256_radioButton">
          <property name="text">
           <string>SHA256</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">hash_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="2" column="2" colspan="2">
         <widget class="QLineEdit" name="accountLineEdit"/>
        </item>
        <item row="6" column="1">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Timer:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Hash Algorithm:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Account Name:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="5" column="2" colspan="2">
         <widget class="QLineEdit" name="secretKeyLineEdit"/>
        </item>
        <item row="11" column="0">
         <widget class="QRadioButton" name="uriRadioButton">
          <property name="text">
           <string>URI</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">manualUri_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Secret Key:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Issuer:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Token Length:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QRadioButton" name="hotp_radioButton">
          <property name="text">
           <string>HOTP</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">tokenType_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="cancelAddEntryButton">
          <property name="text">
           <string>Cancel</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QRadioButton" name="manualRadioButton">
          <property name="text">
           <string>Manual</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">manualUri_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QRadioButton" name="totp_radioButton">
          <property name="text">
           <string>TOTP</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">tokenType_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="QSpinBox" name="timerSpinBox">
          <property name="minimum">
           <number>15</number>
          </property>
          <property name="maximum">
           <number>60</number>
          </property>
          <property name="singleStep">
           <number>15</number>
          </property>
          <property name="value">
           <number>30</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Token Type:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="12" column="2" colspan="2">
         <widget class="QLineEdit" name="uriLineEdit"/>
        </item>
        <item row="8" column="2">
         <widget class="QRadioButton" name="sha1_radioButton">
          <property name="text">
           <string>SHA1</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">hash_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="comfirmAddEntryButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2" colspan="2">
         <widget class="QLineEdit" name="issuerLineEdit"/>
        </item>
        <item row="10" column="2">
         <widget class="QRadioButton" name="sha512_radioButton">
          <property name="text">
           <string>SHA512</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">hash_buttonGroup</string>
          </attribute>
         </widget>
        </item>
        <item row="12" column="1">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>URI:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="7" column="2">
         <widget class="QSpinBox" name="tokenLengthSpinBox">
          <property name="minimum">
           <number>4</number>
          </property>
          <property name="maximum">
           <number>8</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>6</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>591</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>enterPassField</sender>
   <signal>returnPressed()</signal>
   <receiver>enterPassButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newPasswordField</sender>
   <signal>returnPressed()</signal>
   <receiver>confirmPasswordField</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>confirmPasswordField</sender>
   <signal>returnPressed()</signal>
   <receiver>enterNewPassButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="tokenType_buttonGroup"/>
  <buttongroup name="manualUri_buttonGroup"/>
  <buttongroup name="hash_buttonGroup"/>
 </buttongroups>
</ui>
